src/components/App/App.js

```diff
@@ -1,56 +1,56 @@
 import "./App.css";
 import { useState, useEffect } from "react";
-// import movieData from "../../mockMovies";
 import Header from "../Header/Header";
 import MainContentWrapper from "../MainContentWrapper/MainContentWrapper";
 import MovieCards from "../MovieCards/MovieCards";
 // import SingleMovieDetails from "../SingleMovieDetails/SingleMovieDetails";
-import { getAllMovies } from "../../apiCalls/apiCalls";
+import { getAllMovies, getIndividualMovie } from "../../apiCalls/apiCalls";
+import SingleMovieDetails from "../SingleMovieDetails/SingleMovieDetails";
 
 function App() {
   const [fullData, setFullData] = useState([]);
   const [filteredData, setFilteredData] = useState([]);
-  // const [singleMovieDetail, setSingleMovieDetail] = useState();
-  // const [selectedMovie, setSelectedMovieData] = useState( {} );
+  const [singleMovieDetail, setSingleMovieDetail] = useState();
+  const [selectedMovie, setSelectedMovie] = useState(null);
 
   // setSelectedMovieData(fullData.movies.find(movie => movie.id === selectedMovie.id));
 
-    /*
-    In MainContentWrapper we need to add functionality to grab an ID from a clicked movie - use onClick, which would be a part of each card
-    */
-
   useEffect(() => {
-    getAllMovies()
-    .then((data) => {
+    getAllMovies().then((data) => {
       setFullData(data);
       setFilteredData(
         data.movies.map((movie) => {
-        return {
-          id: movie.id,
-          title: movie.title,
-          image: movie.backdrop_path,
-          key: movie.id
-         };
+          return {
+            id: movie.id,
+            title: movie.title,
+            image: movie.backdrop_path,
+            key: movie.id,
+          };
         })
       );
     });
     // .catch((error) => console.log(error.message));
   }, []);
 
-
-
-
-
-
-
-
-
   return (
     <div className="App">
       <Header />
       <MainContentWrapper />
-      <MovieCards filteredData={filteredData} />
-
+      {!selectedMovie && (
+        <MovieCards
+          filteredData={filteredData}
+          setSelectedMovie={setSelectedMovie}
+        />
+      )}
+      {selectedMovie && (
+        <SingleMovieDetails
+          selectedMovie={selectedMovie}
+          setSelectedMovie={setSelectedMovie}
+          singleMovieDetail={singleMovieDetail}
+          setSingleMovieDetail={setSingleMovieDetail}
+          getIndividualMovie={getIndividualMovie}
+        />
+      )}
```

Comment:
Nice conditional logic here!   As you're passing props, consider if each component needs all the props you're passing it.   Also, you _can_ pass the setter functions that you get from setState down to a child component to invoke there, however - its often better to make sure state is only ever being updated from within the component it lives in.  Following that guideline might mean you create some handler functions that invoke these setter functions, and pass the _handlers_ down to the child components instead.